chains/modules/embeddings.py

```diff
@@ -1,11 +1,21 @@
-from langchain.embeddings.huggingface import HuggingFaceEmbeddings
-
+from langchain.embeddings import HuggingFaceEmbeddings
+from ImageBind import data
+import torch
+from ImageBind.models import imagebind_model
+from ImageBind.models.imagebind_model import ModalityType
 from typing import Any, List
 
+device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+
+
 
-class MyEmbeddings(HuggingFaceEmbeddings):
+class MyEmbeddings(HuggingFaceHubEmbeddings):
```

Comment:
How about inherit (BaseModel, Embeddings)? since it is customized embedding instead of huggingface one, so it may be better to inherit a base class?